### PR TITLE
RAW advanced processing

### PR DESCRIPTION
**Pull request by demoulinv**
## Description
Expose some parameters dedicated to raw decoding and DCP color profile application.

## Features list
libraw parameters: highlight Mode, demosaicking algo
White Balancing: Before (libraw) or after (AV) demosaicking
DCP parameter: apply Color Matrix only
Enable reading of images containing raw data in a non raw format (exr)
Add option to select the output color space of a RS panorama in panorama post processing node.

### CODE REVIEW

File: src/aliceVision/image/io.cpp [hunk 1]
```diff
@@ -436,9 +437,39 @@ void readImage(const std::string& path,
...
+                cam_mul = "{1024, 1024, 1024, 1024}";
+                ALICEVISION_LOG_WARNING("[readImage]: cam_mul metadata not available, the openImageIO version might be too old (>= 2.4.5.0 requested for dcp management).");
+            }
+
+            std::vector<float> v_mult;
+            size_t last = 1;
+            size_t next = 1;
+            while ((next = cam_mul.find(",", last)) != std::string::npos)
+            {
+                v_mult.push_back(std::stof(cam_mul.substr(last, next - last).c_str()));
```
Comment (fabiencastan) [inline src/aliceVision/image/io.cpp:460]: no c_str

File: src/aliceVision/image/io.cpp [hunk 2]
```diff
@@ -456,20 +487,21 @@ void readImage(const std::string& path,
...
-            configSpec.attribute("raw:HighlightMode", 1); // unclip
+            configSpec.attribute("raw:HighlightMode", imageReadOptions.highlightMode);
+            configSpec.attribute("raw:Demosaic", imageReadOptions.demosaicingAlgo);
         }
         else if (imageReadOptions.rawColorInterpretation == ERawColorInterpretation::LibRawNoWhiteBalancing)
         {
             configSpec.attribute("raw:auto_bright", 0); // disable exposure correction
             configSpec.attribute("raw:use_camera_wb", 0); // white balance correction
-            configSpec.attribute("raw:use_camera_matrix", 0); // do not use embeded color profile if any
+            configSpec.attribute("raw:use_camera_matrix", 1); // do not use embeded color profile if any except for dng files
```
Comment (fabiencastan) [inline src/aliceVision/image/io.cpp:497]:
```suggestion
            configSpec.attribute("raw:use_camera_matrix", 1); // do not use embeded color profile if any, except for dng files
```

File: src/aliceVision/image/io.cpp [hunk 3]
```diff
@@ -456,20 +487,21 @@ void readImage(const std::string& path,
...
             configSpec.attribute("raw:ColorSpace", "Linear"); // use linear colorspace with sRGB primaries
         }
         else if (imageReadOptions.rawColorInterpretation == ERawColorInterpretation::LibRawWhiteBalancing)
         {
             configSpec.attribute("raw:auto_bright", 0); // disable exposure correction
             configSpec.attribute("raw:use_camera_wb", 1); // white balance correction
-            configSpec.attribute("raw:use_camera_matrix", 0); // do not use embeded color profile if any
+            configSpec.attribute("raw:use_camera_matrix", 1); // do not use embeded color profile if any except for dng files
```
Comment (fabiencastan) [inline src/aliceVision/image/io.cpp:504]:
```suggestion
            configSpec.attribute("raw:use_camera_matrix", 1); // do not use embeded color profile if any, except for dng files
```

File: src/aliceVision/image/io.cpp [hunk 4]
```diff
@@ -547,25 +594,61 @@ void readImage(const std::string& path,
...
+    {
+        ALICEVISION_LOG_INFO("Source image is in a raw color space and must be converted into " << imageReadOptions.workingColorSpace << ".");
+        ALICEVISION_LOG_INFO("Check if a DCP profile is available in the metadata to be applied.");
+        if (inBuf.spec().nchannels < 3)
+        {
+            ALICEVISION_THROW_ERROR("A DCP profile cannot be applied on an image containing less than 3 channels.");
+        }
+
+        int width, height;
+        std::map<std::string, std::string> imageMetadata = getMapFromMetadata(readImageMetadata(path, width, height));
```
Comment (fabiencastan) [inline src/aliceVision/image/io.cpp:627]:
```suggestion
        const std::map<std::string, std::string> imageMetadata = getMapFromMetadata(readImageMetadata(path, width, height));
```

File: src/aliceVision/image/io.cpp (hunk 1 above)
Comment (fabiencastan) [inline src/aliceVision/image/io.cpp:460]:
```suggestion
                v_mult.push_back(std::stof(cam_mul.substr(last, next - last)));
```

File: src/aliceVision/image/io.cpp [hunk 5]
```diff
@@ -436,9 +437,39 @@ void readImage(const std::string& path,
...
+
+            std::vector<float> v_mult;
+            size_t last = 1;
+            size_t next = 1;
+            while ((next = cam_mul.find(",", last)) != std::string::npos)
+            {
+                v_mult.push_back(std::stof(cam_mul.substr(last, next - last).c_str()));
+                last = next + 1;
+            }
+            v_mult.push_back(std::stof(cam_mul.substr(last, cam_mul.find("}", last) - last).c_str()));
```
Comment (fabiencastan) [inline src/aliceVision/image/io.cpp:463]:
```suggestion
            v_mult.push_back(std::stof(cam_mul.substr(last, cam_mul.find("}", last) - last)));
```

File: src/aliceVision/image/io.cpp (hunk 1 above)
Comment (demoulinv) [reply src/aliceVision/image/io.cpp:460]: Done

**Review by fabiencastan — APPROVED** [review]

**Pull request merged**